### PR TITLE
Add basic cursor-based pagination to mds-transactions-service & mds-transactions-api

### PR DESCRIPTION
**Pull request by avatarneil**
## 📚 Purpose
This PR adds basic cursor based pagination functionality to the `mds-transactions-service` and `mds-transactions-api`. There is a pretty robust and extensive test suite for the *service* functionality in this PR, but there are not tests for the *api* yet, due to me trying to keep this PR from getting even larger and more unwieldy. I promise I'll have an API test suite Soon™️!

This PR also adds a new ForceType decorator to `mds-repository` to use in cases where the generated `reflect-metadata` from a given package _cough_ TypeORM for example _cough_ is incorrect. Some libraries, such as our typeorm pagination library, require *accurate* runtime type reflections.

## 📦 Impacts:
- mds-transactions-api
- mds-transactions-service

## ✅ PR Checklist
- [x] Manually test API

### CODE REVIEW

File: helm/mds/templates/deployment.yaml
Comment (avatarneil) [inline helm/mds/templates/deployment.yaml:119]: This is the beginning of some renaming I'd like to do, we don't typically use plurality for our service/api names. So this should really be the `mds-transaction-{api | service}`, not `mds-transactions-{api | service}`.

File: packages/mds-transactions-api/handlers/get-transactions.ts
Comment (avatarneil) [inline packages/mds-transactions-api/handlers/get-transactions.ts:166]: Note that the API does not currently handle error throws especially well. I'll write a global error handling middleware in another PR.

File: packages/mds-transactions-service/repository/index.ts
Comment (avatarneil) [inline packages/mds-transactions-service/repository/index.ts:288]: I added these because test isolation (even within a given file 🙀 ) is super nice, and makes the tests *vastly* more readable and maintainable. No side effects for you *smacks with a stick*!

File: packages/mds-transactions-service/repository/index.ts
Comment (janedotx) [inline packages/mds-transactions-service/repository/index.ts:96]: I must confess I'm not entirely sure what this is doing.

File: packages/mds-transactions-service/tests/transaction-operations.spec.ts
Comment (janedotx) [inline packages/mds-transactions-service/tests/transaction-operations.spec.ts:76]: "non-existEnt"

File: packages/mds-transactions-service/repository/index.ts
Comment (avatarneil) [reply packages/mds-transactions-service/repository/index.ts:96]: It's just turning the `start_timestamp` and `end_timestamp` values into a clause that TypeORM can use for temporal bounds, so if there are both you want your timestamps to be sammiched between them, if you only have a start then you want it to be `>`, and if you only have an end then you want it to be `<`. Is that any more clear?

File: packages/mds-transactions-api/handlers/get-transactions.ts
Comment (cjlynch12) [inline packages/mds-transactions-api/handlers/get-transactions.ts:21]: why the difference in case for these?

File: packages/mds-transactions-api/handlers/get-transactions.ts
Comment (cjlynch12) [inline packages/mds-transactions-api/handlers/get-transactions.ts:43]: is there a more descriptive var other than `x`? Maybe `option_key`?

Comment (avatarneil) [reply packages/mds-transactions-api/handlers/get-transactions.ts:43]: Totes

File: packages/mds-transactions-service/repository/index.ts
Comment (janedotx) [reply packages/mds-transactions-service/repository/index.ts:96]: 👍

File: packages/mds-transactions-service/repository/index.ts
Comment (janedotx) [reply packages/mds-transactions-service/repository/index.ts:288]: Might be a good KTLO project to add something similar to the template. I've been writing my tests in such a way that I don't need to dump the tables in between each group of tests, but this would be nice to have.

File: packages/mds-transactions-service/tests/transactions.spec.ts
Comment (janedotx) [inline packages/mds-transactions-service/tests/transactions.spec.ts:298]: Isn't this `expect` always going to fail, since 'did not happen' is never going to be 'happened', or am I misunderstanding how expect and toBe work?

File: packages/mds-transactions-service/tests/transactions.spec.ts
Comment (janedotx) [inline packages/mds-transactions-service/tests/transactions.spec.ts:262]: Do you still need this?

File: packages/mds-transactions-service/repository/index.ts
Comment (drtyh2o) [reply packages/mds-transactions-service/repository/index.ts:288]: Another suggestion... put this at the beginning of each of these methods....
```ts
if (process.env.NODE_ENV !== 'test') {
   throw new Error(`This method is only supported when executing tests`)
}
```
It seems like it might be possible to add a generic method to the base `ReadWriteRepository` class in @mds-repository that took an entity as a parameter and did this. So you could just call `deleteAll(TransactionStatusEntity)` or something like that.

Finally, this command would return the table to an even more fresh condition:
```sql
TRUNCATE ${repository.metadata.tableName} RESTART IDENTITY;
```

File: packages/mds-transactions-service/repository/entities/transaction-entity.ts
Comment (drtyh2o) [inline packages/mds-transactions-service/repository/entities/transaction-entity.ts:43]: Are you going to disconnect the `TransactionEntityModel` from the `TransactionDomainModel` so that `@ForceType` isn't required?

File: packages/mds-transactions-service/client/index.ts
Comment (drtyh2o) [inline packages/mds-transactions-service/client/index.ts:23]: `.env` files take note.

File: packages/mds-transactions-service/tests/transaction-operations.spec.ts
Comment (drtyh2o) [inline packages/mds-transactions-service/tests/transaction-operations.spec.ts:31]: This assumes DB is "empty" initially. `beforeEach/afterAll` would be certain.

File: packages/mds-transactions-service/tests/transactions.spec.ts
Comment (drtyh2o) [reply packages/mds-transactions-service/tests/transactions.spec.ts:298]: Yeah, I think the preferred style here is something like:
```ts
expect(async () => {
  TransactionServiceClient.getTransactions({ limit: 10000 })
}).rejects.toThrow()
```

File: packages/mds-transactions-api/handlers/get-transactions.ts
Comment (drtyh2o) [inline packages/mds-transactions-api/handlers/get-transactions.ts:65]:
```suggestion
        const isDirection = (value: unknown): value is SORT_DIRECTION => typeof value === 'string' && SORT_DIRECTIONS.includes(value as SORT_DIRECTION)
```

File: packages/mds-transactions-service/@types/index.ts
Comment (drtyh2o) [inline packages/mds-transactions-service/@types/index.ts:116]: If you don't prefer the plural/singular thing, you can have a type/value of the same name and TS figures out which you need based on context, so you can have:
```ts
export const SORTABLE_COLUMN = ['timestamp'] as const
export type SORTABLE_COLUMN = typeof SORTABLE_COLUMN[number]
```
Also, you can use this format for the const assertion
```ts
export const SORT_DIRECTIONS = <const>['ASC', 'DESC']
```

File: packages/mds-transactions-service/tests/transactions.spec.ts
Comment (avatarneil) [reply packages/mds-transactions-service/tests/transactions.spec.ts:298]: Yup, I need to go and switch this test over to the better styling

File: packages/mds-transactions-service/tests/transactions.spec.ts
Comment (avatarneil) [reply packages/mds-transactions-service/tests/transactions.spec.ts:262]: Nope

File: packages/mds-transactions-service/@types/index.ts
Comment (avatarneil) [reply packages/mds-transactions-service/@types/index.ts:116]: Ooh, that's news to me! Very fancy, will try it out

File: packages/mds-transactions-service/repository/entities/transaction-entity.ts
Comment (avatarneil) [reply packages/mds-transactions-service/repository/entities/transaction-entity.ts:43]: Yeah, I'm gonna do that. Will still keep `@ForceType` around in `mds-repository` though, could certainly be useful at some point.

File: packages/mds-transactions-service/repository/entities/transaction-entity.ts
Comment (cjlynch12) [inline packages/mds-transactions-service/repository/entities/transaction-entity.ts:28]: just curious - why the change on these?

**Review by cjlynch12 — APPROVED** [review]
LGTM pending and outstanding comments and another +1 from someone else.

File: packages/mds-transactions-service/repository/index.ts
Comment (avatarneil) [reply packages/mds-transactions-service/repository/index.ts:286]: Good idea for the safeguard @drtyh2o and +1 to the KTLO idea @janedotx. I'm going to add the safeguard for now, and will look to moving this to the base class in a subsequent PR

File: packages/mds-transactions-service/tests/transaction-operations.spec.ts
Comment (avatarneil) [reply packages/mds-transactions-service/tests/transaction-operations.spec.ts]: Resolved in 2bc23236f78585cb9f24ea4ed807c4659a4aa2b5

File: packages/mds-transactions-service/repository/entities/transaction-entity.ts
Comment (avatarneil) [reply packages/mds-transactions-service/repository/entities/transaction-entity.ts:28]: See https://github.com/lacuna-tech/mds-core/pull/542#discussion_r581012236, essentially the reflection generation *is* correct if we specify the primitives directly or through a basic type alias, but not when we're using indexed access types. In my sleuthing around other TypeORM projects, I have yet to find any examples where they are deriving the Entity type from any other complex types (even in DDD projects); we were doing something which made some sense, but there are Reasons™️ why people don't do it. And, FWIW, any type discontinuities ought to be caught within the `mappers.ts` file regardless.

Comment (cjlynch12) [reply packages/mds-transactions-service/repository/entities/transaction-entity.ts:28]: Ahhh gotcha gotcha, missed that connection.
Makes sense 👍

**Review by janedotx — APPROVED** [review]

**Pull request merged**
